### PR TITLE
Add feature which generates Google Cloud Build Configuration file

### LINKED ISSUE #1281: Add feature which generate Cloud Build Configuration file
See: https://github.com/micronaut-projects/micronaut-starter/issues/1280

See: https://cloud.google.com/build

### PR DESCRIPTION
**Pull request by wetted**
closes #1281

### CODE REVIEW

**Comment by wetted** [comment]
I am having some Docker issues smoke testing gradle/maven projects using `cloud-build-local`, but I can't successfully run `./gradlew dockerBuild` either, and it looks like a problem with me running on Apple silicon. I'm investigating, but I believe the code for this feature is correct.

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/gcp/templates/cloudBuild.rocker.raw [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# [START cloudbuild_gradle]
+steps:
+  - name: gradle:7.4.2-jdk@jdkVersion.majorVersion()
+    entrypoint: gradle
+    args: ["test"]
+  - name: gradle:7.4.2-jdk@jdkVersion.majorVersion()
+    entrypoint: gradle
+    args: ["assemble"]
+  - name: gcr.io/cloud-builders/docker
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/gcp/templates/cloudBuild.rocker.raw]: I recommend removing the step to containerize our application. 

Moreover, I think our recommended approach is to containerize the application using Gradle is to use the [Micronaut Gradle Plugin docker related tasks](https://micronaut-projects.github.io/micronaut-gradle-plugin/latest/#_docker_support)

cc @alvarosanchez

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/gcp/templates/cloudBuild.rocker.raw [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+# [START cloudbuild_maven]
+steps:
+  - name: maven:3-jdk-@jdkVersion.majorVersion()
+    entrypoint: mvn
+    args: ["test"]
+  - name: maven:3-jdk-@jdkVersion.majorVersion()
+    entrypoint: mvn
+    args: ["package", "-Dmaven.test.skip=true"]
+  - name: gcr.io/cloud-builders/docker
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/gcp/templates/cloudBuild.rocker.raw]: I recommend removing the step to containerize our application. 

Moreover, I think our recommended approach is to containerize the application using Maven is to use the [Micronaut Maven Plugin Docker packaging options](https://micronaut-projects.github.io/micronaut-maven-plugin/latest/examples/package.html#building_jvm_based_docker_images)

cc @melix

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/gcp/templates/cloudBuild.rocker.raw [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    } else if (buildTool.isGradle()) {
+
+# [START cloudbuild_gradle]
+steps:
+  - name: gradle:7.4.2-jdk@jdkVersion.majorVersion()
+    entrypoint: gradle
+    args: ["test"]
+  - name: gradle:7.4.2-jdk@jdkVersion.majorVersion()
+    entrypoint: gradle
+    args: ["assemble"]
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/gcp/templates/cloudBuild.rocker.raw]: I think we should have a single step which runs the `build` task.

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by sdelamo** [comment]
@wetted is this ready for review or should be in Draft still?

**Pull request merged**